### ISSUE TITLE
[Firefox] web browser : inactive seed or client tab cause a download or upload limit (16.4 kB/s)

### ISSUE THREAD

**Issue by S0llarCode** [webtorrent/webtorrent#2370]
**What version of this package are you using?**
webtorrent.min.js 1.8.29 (latest)
sw.min.js 1.8.29 (latest)

**What operating system, Node.js, and npm version?**
Manjaro Linux 21.3.7 (Gnome)
Firefox 103.0.2 (64 bits)
node v18.7.0
npm 8.17.0

**What happened?**
I noticed that the download slows down when I am not on the tab and returns to normal when I return.

When testing to seeder only from a web browser and not by node, I noticed that the download between the seed tab and the client tab was 16.4 kB/s maximum.

Then I just noticed that if I open a window for the seed tab and another window for the client tab, the download is at its maximum download potential.

I conclude that if one of the tabs (seeder or client) is not active (that we are watching), the transfer is limited to 16.4 kB/s.

[What is an Active tab ?](https://www.computerhope.com/jargon/a/active-tab.htm)

**What did you expect to happen?**
That one can download or upload a file at its maximum download potential without staying on the tab, without the tab being active.

**Are you willing to submit a pull request to fix this bug?**
No. I don't know if this is related to the web browser and/or webtorrent.

**Would you like to do a test?**
I simply created two versions of the [index.html.txt](https://github.com/webtorrent/webtorrent/files/9527625/index.html.txt) file. One where I added the local tracker to the webtorrent client and the other where I changed the magnet so that it only contacts the local tracker.

So, basically, we have a page that downloads the file and seeds it on the local tracker and another one that only uses the local tracker and the seeds on it to download the file.

On Linux, execute : ```touch setup.sh && chmod +x setup.sh```

setup.sh
```
#!/bin/bash
main_dir='webtorrent_testing_seeding'
mkdir "$main_dir"
cd "$main_dir"

npm init -y
npm i express webtorrent

curl -LJO "https://github.com/webtorrent/webtorrent/files/9527625/index.html.txt"
cp index.html.txt local_tracker_client.html
# local_tracker_client.html, line 87
# replace	const torrentId = 'https://webtorrent.io/torrents/sintel.torrent'
# to		const torrentId = 'magnet:?xt=urn:btih:08ada5a7a6183aae1e09d831df6748d566095a10&tr=ws%3A%2F%2F127.0.0.1%3A8081'
sed -i '87s/https:\/\/webtorrent.io\/torrents\/sintel.torrent/magnet:?xt=urn:btih:08ada5a7a6183aae1e09d831df6748d566095a10\&tr=ws%3A%2F%2F127.0.0.1%3A8081/' local_tracker_client.html

mv index.html.txt default_and_local_tracker_client.html
# default_and_local_tracker_client.html, line 105
# replace client.add(torrentId, function (torrent) { 
# to 		client.add(torrentId, {announce: ['ws://127.0.0.1:8081']}, function (torrent) {
sed -i '105s/torrentId, /torrentId, {announce: ["ws:\/\/127.0.0.1:8081"]}, /' default_and_local_tracker_client.html

curl -O "https://cdn.jsdelivr.net/npm/webtorrent@latest/sw.min.js"
```

Create ```webtracker.js``` file in working folder (webtorrent_testing_seeding)
By executing for example : ```touch webtorrent_testing_seeding/webtracker.js```

webtracker.js
```
const Server = require('bittorrent-tracker').Server
const hostname = "127.0.0.1"
const port = 8081

const log_h = '[WebTracker]:'

const server = new Server({
  udp: false, 
  http: false, 
  ws: true, 
  stats: false, 
  trustProxy: false
})

server.on('error', function (err) {
  console.log(log_h, err)
})
server.on('warning', function (err) {
  console.log(log_h, err)
})

server.on('start', function (addr) {
  console.log(log_h, 'Got start message from', addr)
})
server.on('complete', function (addr) {
  console.log(log_h, 'Got complete message from', addr)
})
server.on('update', function (addr) {
  console.log(log_h, 'Got update message from', addr)
})
server.on('stop', function (addr) {
  console.log(log_h, 'Got stop message from', addr)
})

server.on('listening', function () {
  const wsAddr = server.ws.address()
  const wsHost = wsAddr.address !== '::' ? wsAddr.address : 'localhost'
  const wsPort = wsAddr.port
  console.log(log_h, `WebSocket tracker: ws://${wsHost}:${wsPort}`)
})

server.listen(port, hostname, function () {
  console.log(log_h, 'Start listening...')
})
```

Create ```webserver.js``` file in working folder (webtorrent_testing_seeding)
By executing for example : ```touch webtorrent_testing_seeding/webserver.js```

webserver.js
```
const express = require('express')
const app = express()
const port = 8080

app.use('/sw.min.js', express.static(__dirname + '/sw.min.js'))
app.use('/default_and_local_tracker_client.html', express.static(__dirname + '/default_and_local_tracker_client.html'))
app.use('/local_tracker_client.html', express.static(__dirname + '/local_tracker_client.html'))

app.listen(port)

console.log(`[WebServer]: http://localhost:${port}/default_and_local_tracker_client.html`)
console.log(`[WebServer]: http://localhost:${port}/local_tracker_client.html`)
```

Create ```app.js``` file in working folder (webtorrent_testing_seeding)
By executing for example : ```touch webtorrent_testing_seeding/app.js```

app.js
```
async function runWebTracker() {
  require('./webtracker.js')
}
async function runWebServer() {
  require('./webserver.js')
}
runWebTracker()
runWebServer()
```

To start the test : 

In woking folder webtorrent_testing_seeding
Execute : ```node app.js```

In web browser : http://localhost:8080/default_and_local_tracker_client.html
For more speed, you can wait on the tab until the file is in seed.

In ```new tab``` : http://localhost:8080/local_tracker_client.html
Normally you will find that the download is very slow.
After the connection has stabilized, about 4s, the download rate is 16.4 kB/s.

Return to the seed tab and ```open a new window``` : http://localhost:8080/local_tracker_client.html
If you have like me, in windowed mode, the seed page in the background and the client page in the foreground you will see that the download is normal.

If you have only one of the two tabs active then the download and/or upload is limited, like a sleep mode because one of the two tabs is inactive.

**Comment by ThaUnknown** [comment]
could you test `1.8.26`?

**Comment by S0llarCode** [comment]
**What version of this package are you using?**
[webtorrent.min.js](https://cdn.jsdelivr.net/npm/webtorrent@1.8.26/webtorrent.min.js) `1.8.26`
[sw.min.js](https://raw.githubusercontent.com/webtorrent/webtorrent/v1.8.26/sw.min.js) `1.8.26`

**What happened?**
Same results.

**Comment by ThaUnknown** [comment]
see https://github.com/webtorrent/webtorrent/issues/2191 for fix

duplicate of https://github.com/webtorrent/webtorrent/issues/1568
